client/lib/create-cached-selector/test/index.js

```diff
@@ -0,0 +1,192 @@
+/** @format */
+import { filter } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createCachedSelector, { MixedMap } from '../';
+import warn from 'lib/warn';
+
+jest.mock( 'lib/warn', () => jest.fn() );
+
+describe( 'index', () => {
+	describe( '#createCachedSelector', () => {
+		const post1 = { id: 'id1', text: 'here is post 1', siteId: 'site1' };
+		const post2 = { id: 'id2', text: 'here is post 2', siteId: 'site1' };
+		const post3 = { id: 'id3', text: 'here is post 3', siteId: 'site2' };
+
+		let getSitePosts;
+		let selector;
+		let getDependents;
+
+		beforeEach( () => {
+			selector = jest.fn( ( { posts }, siteId ) => filter( posts, { siteId } ) );
+			getDependents = jest.fn( state => ( { posts: state.posts } ) );
+			getSitePosts = createCachedSelector( { selector, getDependents } );
+		} );
+
+		test( 'should create a function which returns the expected value when called', () => {
+			const state = {
+				posts: {
+					[ post1.id ]: post1,
+					[ post2.id ]: post2,
+					[ post3.id ]: post3,
+				},
+			};
+
+			expect( getSitePosts( state, 'site1' ) ).toEqual( [ post1, post2 ] );
+		} );
+
+		test( 'should cache the result of a selector function', () => {
+			const reduxState = {
+				posts: {
+					[ post1.id ]: post1,
+					[ post2.id ]: post2,
+					[ post3.id ]: post3,
+				},
+			};
+
+			getSitePosts( reduxState, 2916284 );
+			getSitePosts( reduxState, 2916284 );
+
+			expect( selector.mock.calls.length ).toBe( 1 );
+		} );
```

Comment:
A good partner-test here might be one to verify that this gets called twice when the arguments are different. This test has a high risk of false positives on the behavior of the function.

```js
test( 'should call selector when making non-cached calls', () => {
	getSitePosts( reduxState, 2916284 )
	getSitePosts( reduxState, 2916284 + 1 )

	expect( selector.mock.calls.length ).toBe( 2 );
} )
```

Reply:
I think the current test named that `should return the expected value of differing arguments` does this.

I like your test naming better though, I'll switch